account/api/serializers.py

```diff
@@ -1,8 +1,100 @@
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
 from rest_framework import serializers
-from account.models import PhonePrefix, CustomUser
+
+
+from account.models import CustomUser, PhonePrefix, Warehouse
+from account.tokens import account_activation_token
+
 
 class PhonePrefixSerializer(serializers.ModelSerializer):
-     
-     class Meta:
+    class Meta:
         model = PhonePrefix
-        fields = ['prefix_number']
+        fields = "__all__"
+
+    def create(self, validated_data):
+        return PhonePrefix.objects.create(**validated_data)
+
+
+class BranchSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Warehouse
+        fields = "__all__"
+
+    def create(self, validated_data):
+        return Warehouse.objects.create(**validated_data)
+
+
+class RegistrationSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(
+        write_only=True, required=True, style={"input_type": "password"}
+    )
+    phone_prefix = PhonePrefixSerializer(many=False)
+    branch = BranchSerializer(many=False)
+
+    class Meta:
+        model = CustomUser
+        fields = (
+            "email",
+            "password",
+            "first_name",
+            "last_name",
+            "gender",
+            "phone_prefix",
+            "phone",
+            "gov_id_prefix",
+            "gov_id",
+            "pin_code",
+            "birth_date",
+            "branch",
+        )
+
+    def validate(self, attrs):
+        password = attrs.get("password")
+        password_confirm = self.context.get("request").data.get("password_confirm")
+        if password and password_confirm and password != password_confirm:
+            raise serializers.ValidationError({"password": "Passwords do not match."})
+        return attrs
+
+    def create(self, validated_data):
+        phone_prefix_data = validated_data.pop("phone_prefix")
+        warehouse_data = validated_data.pop("branch")
+        phone_prefix = PhonePrefix.objects.create(**phone_prefix_data)
+        warehouse = Warehouse.objects.create(**warehouse_data)
+        user = CustomUser.objects.create_user(
```

Comment:
Consider that, for each user you are going to create new phone prefix, which is not correct. Instead, it should be FK.